From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Wed, 26 Aug 2026 11:26:14 +0200
Subject: fix(dmsquash-live-autooverlay): support grep applet from busybox

busybox's `grep` applet does not support the long form `--count` but it
supports the short form `-c`.

Part of: https://github.com/dracut-ng/dracut/issues/2437
Origin: upstream, https://github.com/dracut-ng/dracut/commit/84d912fc9195a8e572daa2c5a63d359dee66e40c
---
 modules.d/70dmsquash-live-autooverlay/create-overlay.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/70dmsquash-live-autooverlay/create-overlay.sh b/modules.d/70dmsquash-live-autooverlay/create-overlay.sh
index 0eb8948..513549c 100755
--- a/modules.d/70dmsquash-live-autooverlay/create-overlay.sh
+++ b/modules.d/70dmsquash-live-autooverlay/create-overlay.sh
@@ -54,7 +54,7 @@ gatherData() {
     fi
     fullDriveSysfsPath=$(readlink -f "${rootDeviceSysfsPath}/..")
     blockDevice="/dev/${fullDriveSysfsPath##*/}"
-    currentPartitionCount=$(grep --count -E "${blockDevice#/dev/}[0-9]+" /proc/partitions)
+    currentPartitionCount=$(grep -Ec "${blockDevice#/dev/}[0-9]+" /proc/partitions)
 
     freeSpaceStart=$(parted --script "${blockDevice}" unit % print free \
         | awk -v "x=${currentPartitionCount}" '$1 == x {getline; print $1}')
